core-api/src/test/java/com/optimizely/ab/config/audience/AudienceConditionEvaluationTest.java

```diff
@@ -113,8 +113,11 @@ public void invalidMatchCondition() throws Exception {
     @Test
     public void invalidMatch() throws Exception {
         UserAttribute testInstance = new UserAttribute("browser_type", "custom_attribute", "blah", "chrome");
+        String expectedAudience = testInstance.toString();
         assertNull(testInstance.evaluate(null, testUserAttributes));
-        logbackVerifier.expectMessage(Level.WARN, String.format("Audience condition \"%s\" uses an unknown match type. You may need to upgrade to a newer release of the Optimizely SDK", testInstance.toString()));
+        logbackVerifier.expectMessage(Level.WARN,
+            String.format("Audience condition \"%s\" uses an unknown match type. You may need to upgrade to a newer release of the Optimizely SDK",
+                expectedAudience));
```

Comment:
Hmm, by "inline the audience condition" I meant can we do something like
```java
logbackVerifier.expectMessage(Level.WARN, "Audience condition {name:'blah', type='custom_attribute', match='blah', value='chrome'} uses an unknown match type. You may need to upgrade to a newer release of the Optimizely SDK");
```
This way, if `UserAttribute.toString` behaves unexpectedly, we can tell by looking at our test cases.

(Same thing everywhere below)